File: dotnet/src/AI.Sentinel.AspNetCore/Contract/ContractRiskFactor.cs
using System.Text.Json.Serialization;

namespace AI.Sentinel.AspNetCore.Contract;

public sealed class ContractRiskFactor
{
    [JsonPropertyName("code")]
    public string Code { get; set; } = string.Empty;

    [JsonPropertyName("category")]
    public string Category { get; set; } = string.Empty;

    [JsonPropertyName("severity")]
    public string Severity { get; set; } = string.Empty;

    [JsonPropertyName("contribution")]
    public double Contribution { get; set; }

    [JsonPropertyName("confidence")]
    public double Confidence { get; set; }

    [JsonPropertyName("evidenceRef")]
    public string EvidenceRef { get; set; } = string.Empty;

    [JsonPropertyName("explanation")]
    public string Explanation { get; set; } = string.Empty;

    [JsonPropertyName("source")]
    public string Source { get; set; } = string.Empty;
}

File: ai-sentinel-core/src/main/java/dev/aisentinel/core/contract/ContractRiskFactor.java
package dev.aisentinel.core.contract;

import java.util.Objects;

/**
 * Transport-safe risk factor row. Codes are stable enum names; contribution/confidence are finite.
 */
public record ContractRiskFactor(
    String code,
    String category,
    String severity,
    double contribution,
    double confidence,
    String evidenceRef,
    String explanation,
    String source
) {
    public ContractRiskFactor {
        Objects.requireNonNull(code, "code");
        Objects.requireNonNull(category, "category");
        Objects.requireNonNull(severity, "severity");
        if (!Double.isFinite(contribution) || contribution < 0.0 || contribution > 1.0) {
            throw new EvaluationContractException("contribution must be finite in [0,1]");
        }
        if (!Double.isFinite(confidence) || confidence < 0.0 || confidence > 1.0) {
            throw new EvaluationContractException("confidence must be finite in [0,1]");
        }
        evidenceRef = evidenceRef == null ? "" : evidenceRef;
        explanation = explanation == null ? "" : explanation;
        source = source == null ? "" : source;
    }
}
